### ISSUE TITLE
Missing language data for es-ES (intl-displaynames)

### ISSUE THREAD

**Issue by roderickhsiao** [formatjs/formatjs#541]
**Which package?**
intl-displaynames

**Describe the bug**
Will expect `es-ES` in the dist locale data, but seems missing.

https://unpkg.com/browse/@formatjs/intl-displaynames@1.2.0/dist/locale-data/

**Comment by roderickhsiao** [comment]
`tl`  is `fil`

**Comment by longlho** [comment]
I believe it falls back to `es`

**Comment by roderickhsiao** [comment]
The issues that when we do 
```js
import '@formatjs/intl-displaynames/dist/locale-data/es-ES'; 
```
(We are using dynamic import so we wasn't aware on build time)

the code will throw error, it seems like we have to try individual locale to see if the data is there. 
for package like https://github.com/andyearnshaw/Intl.js I think they includes all the combinations.

**Comment by longlho** [comment]
hmm Intl.js does a combo of the locales it supports https://github.com/andyearnshaw/Intl.js/blob/4bd382c5df206f328b35ca23579904f603b84f0b/scripts/utils/locales.js#L41 so it's not really `all`. Doing dynamic import w/o verification is pretty dangerous (especially for aliases like `zh-CN -> zh-Hant-CN`) and there's a whole fallback chain that happens in place (e.g `en-US` -> `en-001` -> `en`). 

I think it'd prob be safer if we provide a locale resolution utility. Would that work?

**Comment by roderickhsiao** [comment]
@longlho yeah, I was checking `cldr-core` https://github.com/unicode-cldr/cldr-core/blob/master/availableLocales.json, and our implementation here follows correctly based on the data provided.

And yeah, I wasn't aware the dangerous part until production outage :(( 

Our temporary solution is to manually clone/modified `pt-BR` from `pt` and `es-ES` from `es`, change `tl` to `fil` as we need the same language code as in react-intl and the locale data here. (and try/catch with `en` fallback), which probably could be solved by the utils (same fallback chain in place).

I was originally thinking just an utils to check if the locale data is there, but I like the idea that returning based on the fallback chain.
